crates/nu-command/src/core_commands/do_.rs

```diff
@@ -106,7 +109,8 @@ impl Command for Do {
             block,
             input,
             call.redirect_stdout,
-            capture_errors || ignore_shell_errors || ignore_program_errors,
+            // TODO remove capture_errors once redirection is fixed
+            call.redirect_stderr || capture_errors,
```

Comment:
I think `call.redirect_stderr` will always be false, so it's meanless

I'd suggest change the line to: `capture_errors || !ignore_all_errors`, so do -i will not redirect stdout, and standard do without -i will capture stderr and have a chance to collect by complete.

Reply:
Doesn't that mean apt and friends will be broken if you stick them in a plain do block? I still fundamentally disagree with the idea of making this stuff sometimes work and sometimes not depending on how you massage it.

Reply:
yeah, but it works for `do -i`, One of `do` and `do -i` must capture stderr.

I think we made a decision that `do -i` should made nushell ignore stderr message(so it can be printed), so plain do should capture stderr and can be catched by `complete` command

Reply:
Refer to comment: https://github.com/nushell/nushell/pull/7204#issuecomment-1337896214

I still want `do` to do something like @theAkito mentioned.
```
> do { sh -c "function bad { echo good; echo bad 1>&2; return 1; }; bad" }            
good
bad

> $env.LAST_EXIT_CODE
1
```
Standard behaviour without complete(This can't be completed, but we can achieve the behavior with `do -i`).

```
do { sh -c "function bad { echo good; echo bad 1>&2; return 1; }; bad" } | complete
┏━━━━━━━━━━━┳━━━━━━┓
┃ stdout    ┃ good ┃
┃           ┃      ┃
┃ stderr    ┃ bad  ┃
┃           ┃      ┃
┃ exit_code ┃ 1    ┃
┗━━━━━━━━━━━┻━━━━━━┛
```
Standard behaviour with complete(this will be ok).

```
do -i { sh -c "function bad { echo good; echo bad 1>&2; return 1; }; bad" } | complete
┏━━━━━━━━━━━┳━━━━━━┓
┃ stdout    ┃ good ┃
┗━━━━━━━━━━━┻━━━━━━┛
```
Ignoring all errors(this will be ok).

Reply:
Yeah, I totally agree that `do { blah } | complete` should work, but I'm saying it shouldn't work at the expense of breaking blah (which is currently the case AFAIK). Maybe this is something that should be moved to an issue about managing redirection with complete so it can be fixed properly?

Reply:
Ok, I agree with you, I'd vote for landing it.

Reply:
Randomly thinking, if `do { blah } | complete` works and complete catch `stderr`, `blah` must redirect `stderr`.

I think ideally what we can reach is make `do { blah }` works without redirecting `stderr`, and `do { blah } | complete` works, but  `blah` must redirecting `stderr`